src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -547,6 +526,48 @@ private bool CheckParameterValueKind(SyntaxNode node, BoundParameter parameter,
             return true;
         }
 
+        private static bool CheckParameterRefEscape(SyntaxNode node, BoundParameter parameter, uint escapeTo, bool checkingReceiver, DiagnosticBag diagnostics)
+        {
+            ParameterSymbol parameterSymbol = parameter.ParameterSymbol;
+            var paramKind = parameterSymbol.RefKind;
+
+            // byval parameters can escape to method's top level.
+            // others can be escape further, unless they are ref-like.
+            if (escapeTo == Binder.ExternalScope)
+            {
+                if (parameterSymbol.RefKind == RefKind.None)
+                {
+                    if (checkingReceiver)
+                    {
+                        Error(diagnostics, ErrorCode.ERR_RefReturnParameter2, parameter.Syntax, parameterSymbol.Name);
+                    }
+                    else
+                    {
+                        Error(diagnostics, ErrorCode.ERR_RefReturnParameter, node, parameterSymbol.Name);
+                    }
+                    return false;
+                }
+
+                if (parameterSymbol?.Type.IsByRefLikeType == true)
```

Comment:
right, but type could probably be (not sure., but better be safe here) . 
Will have to move "?" to after the "Type"

---
In reply to: [134883004](https://github.com/dotnet/roslyn/pull/21334#discussion_r134883004) [](ancestors = 134883004)